### PR TITLE
[23-01-27] 과일장수 (1단계)

### PR DESCRIPTION
**Pull request by Jeong-jeong**
## 문제
<p style="box-sizing: border-box; padding: 0px; margin: 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: 1.8; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.1px; overflow-wrap: break-word; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">과일 장수가 사과 상자를 포장하고 있습니다. 사과는 상태에 따라 1점부터 k점까지의 점수로 분류하며, k점이 최상품의 사과이고 1점이 최하품의 사과입니다. 사과 한 상자의 가격은 다음과 같이 결정됩니다.</p><ul style="box-sizing: border-box; padding: 0px 0px 0px 2.5rem; margin: 1rem 0px 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: inherit; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit;">한 상자에 사과를 m개씩 담아 포장합니다.</li><li style="box-sizing: border-box; padding: 0px; margin: 0.5rem 0px 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit;">상자에 담긴 사과 중 가장 낮은 점수가 p (1 ≤ p ≤ k)점인 경우, 사과 한 상자의 가격은 p * m 입니다.</li></ul><p style="box-sizing: border-box; padding: 0px; margin: 0.75rem 0px 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: 1.8; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.1px; overflow-wrap: break-word; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">과일 장수가 가능한 많은 사과를 팔았을 때, 얻을 수 있는 최대 이익을 계산하고자 합니다.(사과는 상자 단위로만 판매하며, 남는 사과는 버립니다)</p><p style="box-sizing: border-box; padding: 0px; margin: 0.75rem 0px 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: 1.8; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.1px; overflow-wrap: break-word; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">예를 들어,<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem;">k</code><span> </span>= 3,<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem;">m</code><span> </span>= 4, 사과 7개의 점수가 [1, 2, 3, 1, 2, 3, 1]이라면, 다음과 같이 [2, 3, 2, 3]으로 구성된 사과 상자 1개를 만들어 판매하여 최대 이익을 얻을 수 있습니다.</p><ul style="box-sizing: border-box; padding: 0px 0px 0px 2.5rem; margin: 1rem 0px 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: inherit; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit;">(최저 사과 점수) x (한 상자에 담긴 사과 개수) x (상자의 개수) = 2 x 4 x 1 = 8</li></ul><p style="box-sizing: border-box; padding: 0px; margin: 0.75rem 0px 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: 1.8; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.1px; overflow-wrap: break-word; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">사과의 최대 점수<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem;">k</code>, 한 상자에 들어가는 사과의 수<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem;">m</code>, 사과들의 점수<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem;">score</code>가 주어졌을 때, 과일 장수가 얻을 수 있는 최대 이익을 return하는 solution 함수를 완성해주세요.</p><h5 style="box-sizing: border-box; padding: 0px; margin: 1.5rem 0px 0px; color: white; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 15px; line-height: 1.5; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.009em; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">제한사항</h5><ul style="box-sizing: border-box; padding: 0px 0px 0px 2.5rem; margin: 0.75rem 0px 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: inherit; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit;">3 ≤<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem;">k</code><span> </span>≤ 9</li><li style="box-sizing: border-box; padding: 0px; margin: 0.5rem 0px 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit;">3 ≤<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem;">m</code><span> </span>≤ 10</li><li style="box-sizing: border-box; padding: 0px; margin: 0.5rem 0px 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit;">7 ≤<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem;">score</code>의 길이 ≤ 1,000,000<ul style="box-sizing: border-box; padding: 0px 0px 0px 2.5rem; margin: 0.75rem 0px 0px; color: inherit; font: inherit;"><li style="box-sizing: border-box; padding: 0px; margin: 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit;">1 ≤<span> </span><code style="box-sizing: border-box; padding: 2px 6px; margin: 0px 2px 0px 0px; color: rgb(205, 215, 224); font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: Hack, Menlo, Monaco, Consolas, &quot;Courier New&quot;, monospace, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif; overflow-wrap: break-word; border: 1px solid rgb(23, 35, 52); background-color: rgb(32, 43, 61); border-radius: 0.25rem;">score[i]</code><span> </span>≤ k</li></ul></li><li style="box-sizing: border-box; padding: 0px; margin: 0.5rem 0px 0px; color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: 16px; line-height: 1.6; font-family: inherit; letter-spacing: -0.009em; list-style: inherit;">이익이 발생하지 않는 경우에는 0을 return 해주세요.</li></ul><hr style="box-sizing: content-box; padding: 0px; margin: 1.5rem 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: inherit; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; height: 0px; overflow: visible; border-width: 0.0625rem 0px 0px; border-right-style: initial; border-bottom-style: initial; border-left-style: initial; border-color: rgb(23, 35, 52); border-image: initial; border-top-style: solid; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h5 style="box-sizing: border-box; padding: 0px; margin: 1.5rem 0px 0px; color: white; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 15px; line-height: 1.5; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.009em; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">입출력 예</h5>

k | m | score | result
-- | -- | -- | --
3 | 4 | [1, 2, 3, 1, 2, 3, 1] | 8
4 | 3 | [4, 1, 2, 2, 4, 4, 4, 4, 1, 2, 4, 2] | 33

<p style="box-sizing: border-box; padding: 0px; margin: 0.75rem 0px 0px; color: rgb(178, 192, 204); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 16px; line-height: 1.8; font-family: Inter, NotoSansKR, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;맑은 고딕&quot;, &quot;malgun gothic&quot;, 돋움, Dotum, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Noto Color Emoji&quot;; letter-spacing: -0.1px; overflow-wrap: break-word; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(38, 55, 71); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">따라서 (1 x 3 x 1) + (2 x 3 x 1) + (4 x 3 x 2) = 33을 return합니다.</p>

## 푸는데 걸린 시간
20분

## 깨달은 점 (필요한 경우)
index 조작이 필요한 경우, 괜히 용쓴다고 함수형 메서드 쓰려고 하지 말고,
for 문으로 해결하라.

## 스크린샷 (필요할 경우)
image

## 관련 문서 링크 (필요할 경우)
https://school.programmers.co.kr/learn/courses/30/lessons/135808

### CODE REVIEW

File: jeongs/jan/23-01-27-과일장수.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// 과일 장수가 얻을 수 있는 최대 이익 return
+// 한 상자의 가장 낮은 점수 사과 * 사과 m개
+// 남는 사과 고려 x
+// 이익이 발생하지 않는 경우 0 return
+
+// 가능한 많은 사과
+// 가능한 많은 이익 = 한 상자(m)에 최저 점수가 높아야 함
```
Comment (ezzange) [inline jeongs/jan/23-01-27-과일장수.js:7]: 나는 훔친다. 좋은 습관

File: jeongs/jan/23-01-27-과일장수.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    for (let i = 0; i < score.length; i += m) {
+        const box = score.slice(i, i + m);
+
+        if (box.length < m) break;
+        
+        result += (Math.min(...box) * m);
+    }
+    
+    return result;
+}
```
Comment (ezzange) [inline jeongs/jan/23-01-27-과일장수.js:22]: 이렇게 간단하다니 놀랍도다.
어지러우니 나는 잠시 쉬어야겠다.

**Review by ezzange — APPROVED** [review]

File: jeongs/jan/23-01-27-과일장수.js (hunk 1 above)
Comment (Jeong-jeong) [reply jeongs/jan/23-01-27-과일장수.js:7]: 괴도졍

File: jeongs/jan/23-01-27-과일장수.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// 과일 장수가 얻을 수 있는 최대 이익 return
```
Comment (sobaman) [inline jeongs/jan/23-01-27-과일장수.js:1]: 이 집 잘 파네

**Review by sobaman — APPROVED** [review]

**Pull request merged**